src/prommis/examples/cmi_process_flowsheet/cmi_process_flowsheet.py

```diff
@@ -0,0 +1,670 @@
+
+    print(degrees_of_freedom(m))
+
+
+def initialize_system(m):
+    """
+    Initialize system.
+
+    Args:
+        m: pyomo model
+    """
+
+    ### Initialize Feed and propagate state to Dissolution Stage
+    m.fs.FEED.initialize()
+    propagate_state(arc=m.fs.FEED_Diss)
+
+    ### Initialize Dissolution Stage and propagate state to S101
+    m.fs.Dissolution.initialize()
+    propagate_state(arc=m.fs.Diss_S101)
+
+    ### Initialize S101 and propagate state to pH adjustment stage mixer
+    m.fs.S101.initialize()
+    propagate_state(arc=m.fs.S101_AdjMixer)
+
+    # Initialize pH Adjustment feed and propagate state to mixer
+    m.fs.AdjFeed.initialize()
+    propagate_state(arc=m.fs.AdjFeed_AdjMixer)
+
+    # Initialize pH Adjustment mixer and propagate state to reactor
+    m.fs.AdjMixer.initialize()
+    propagate_state(arc=m.fs.AdjMixer_Adjustment)
+
+    # Initialize pH Adjustment reactor and propagate state to precipitation stage mixer
+    m.fs.Adjustment.initialize()
+    propagate_state(arc=m.fs.Adjustment_PrecipMixer)
+
+    # Initialize precipitation stage feed and propagate state to precipitation stage mixer
+    m.fs.PrecipFeed.initialize()
+    propagate_state(arc=m.fs.PrecipFeed_PrecipMixer)
+
+    # Initialize precipitation stage feed and propagate state to precipitation stage reactor
+    m.fs.PrecipMixer.initialize()
+    propagate_state(arc=m.fs.PrecipMixer_Precipitation)
+
+    # Initialize precipitation stage reactor
+    m.fs.Precipitation.initialize()
+    propagate_state(arc=m.fs.Precipitation_S102)
+
+    ### Initialize S102
+    m.fs.S102.initialize()
+
+    # Initialize precipitation stage calcinator
+    m.fs.Calcination.initialize()
+
+
+def solve_system(m, tee=False):
+    """
+    Args:
+        m: pyomo model
+        solver: optimization solver
+        tee: boolean indicator to stream IPOPT solution
+    """
+    # Solve flowsheet
+    solver_obj = SolverFactory(
+        "ipopt",
+        options={
+            "nlp_scaling_method": "user-scaling",
+            "tol": 1e-6,
+            "max_iter": 1000,
+        },
+    )
+
+    results = solver_obj.solve(m, tee=tee)
+    assert_optimal_termination(results)
+
+    return results
+
+
+def display_results(m):
+    """
+    Print key flowsheet outputs.
+
+    Args:
+        m: pyomo model
+    """
+
+    print(
+        "Molar flowrate of REPM entering process: {:0.2f} mol/s".format(
+            value(m.fs.FEED.flow_mol_phase_comp[0, "Sol", "Nd2Fe14B"])
+        )
+    )
+
+    m.fs.FEED.report()
+    m.fs.Dissolution.report()
+    m.fs.S101.report()
+    m.fs.AdjFeed.report()
+    m.fs.AdjMixer.report()
+    m.fs.Adjustment.report()
+    m.fs.PrecipFeed.report()
+    m.fs.PrecipMixer.report()
+    m.fs.Precipitation.report()
+    m.fs.S102.report()
+
+    print(
+        "\nMolar Flowrate of product Nd2O3 recovered: {:0.5f} mol/s".format(
+            value(m.fs.Calcination.flow_mol_comp_product[0, "Nd"])
+        )
+    )
+
+
+main()
```

Comment:
Modify so that `main()` runs when the module is executed:

```suggestion
if __name__ == "__main__":
    main()
```

Reply:
This has been added.